### PR TITLE
Rename `extend` to `utilsExtend`

### PR DESCRIPTION
**Pull request by seanpdoyle**
Running a project's CI with `phantomjs@~1.9.0` raises the following
error:

```
Exception while running qunit-bdd `before` hook:
`undefined` is not a function
  (evaluating `extend['default'].bind(Model, null)`)
```

`extend` seems to be a pre-existing variable (and will probably be a
keyword in future browsers). Therefore `extend['default']` is
`undefined`.

To work around this, export the `utils/extend` to a different name.

### CODE REVIEW

**Comment by seanpdoyle** [comment]
On further inspection, the issue seems to be with `.bind()`.

Either way, `extend` will probably be a reserved word, so I'll leave this open to discussion.

**Comment by blimmer** [comment]
Sounds like you got the `.bind` issue figured out - I'm not sure if we should change this because it might be a problem in the future.

**Comment by seanpdoyle** [comment]
@blimmer could you elaborate on the details of why this rename would be a problem in the future?

**Comment by blimmer** [comment]
Sorry, I think there was a miscommunication. Let me rephrase - I don't think there's a strong case to make this change just because we think it could maybe be a reserved word in the future.
